fastcrypto-vdf/src/class_group/mod.rs

```diff
@@ -101,8 +101,7 @@ impl QuadraticForm {
 
     /// Compute the discriminant `b^2 - 4ac` for this quadratic form.
     pub fn discriminant(&self) -> Discriminant {
-        Discriminant::try_from(self.b.pow(2) - ((&self.a * &self.c) << 2))
-            .expect("The discriminant is checked in the constructors")
+        Discriminant::from_trusted_bigint(self.b.pow(2) - ((&self.a * &self.c) << 2))
```

Comment:
how do we ensure this will always be invoked from trusted?

Reply:
This was fixed in #787